tasks/winbuildscripts/Generate-Pyzip.ps1

```diff
@@ -61,45 +86,79 @@ if($maj -eq "2") {
 }
 elseif ($maj -eq "3") {
     $Env:PATH="$Env:PATH;c:\tools\msys64\mingw64\bin"
-    $p = Start-Process $dlfullpath -ArgumentList "/quiet DefaultJustForMeTargetDir=c:\pythonroot TARGETDIR=c:\pythonroot AssociateFiles=0 Include_doc=0 Include_launcher=0 Include_pip=0 Include_tcltk=0" -Wait -Passthru
 
-    if ($p.ExitCode -ne 0) {
-        Write-Host -ForegroundColor Red ("Failed to install Python, exit code 0x{0:X}" -f $p.ExitCode)
-        exit 1
+    $needInstall = $false
+    $needUninstall = $false
+    try {
+        $installedPythonVersion = python --version
+        $installedPythonVersion = [regex]::match($installedPythonVersion,'Python (\d.\d.\d)').Groups[1].Value
+        $installedMaj, $installedMin, $installedPatch = $installedPythonVersion.Split(".")
+    
+        Write-Host -ForegroundColor Green "Detected installed Python $installedMaj $installedMin $installedPatch"
+
+        if ($installedMaj -ne "3") {
+            Write-Host -ForegroundColor Red "Invalid Python $installedPythonVersion installation detected"
+            exit 1   
+        }
+
+        if (!(($installedMin -eq $min) -and ($installedPatch -eq $patch))) {
```

Comment:
Given what we're trying to do here, I think I'd uninstall even if it's the right version, and reinstall.
There's no guarantee the base version (from `buildimages`) is installed with the same options/directory layout, and so may not have all the right contents.
I'd suggest just nuke it and start fresh.